### ISSUE TITLE
[Small Inconpatability] Issues with the Moon

### ISSUE THREAD

**Issue by SpookyTofu** [Povstalec/StellarView#30]
Been trying to make a space themed mod pack and I wanted to use a resource pack that turns the celestial bodies into 3D cubes instead of 2D squares. The resource pack is called "Cubic Sun & Moon" by JoeFly. 
Link to the Resource pack. https://modrinth.com/resourcepack/cubic-sun-moon

The problem come in when both the resource pack and Stellar view are used together. It's nothing to big. The model for the sun replaces the vanilla sun perfectly, but the moon can't replace the vanilla moon.

Video Clip of StellarView disabled.
https://github.com/user-attachments/assets/b843830a-a8d5-450a-aaea-44bb224d03bd

Video Clip of StellarView enabled.
https://github.com/user-attachments/assets/f08ac1f6-ddba-4378-8286-ff34f7d1646a

hopefully this can be an easy fix since the sun works fine. Only slight changes to the moon would be needed.

**Comment by Povstalec** [comment]
The Vanilla Minecraft Moon uses only one texture, just like the Sun, because it just so happens that Mojang chose a location for the Moon in the sky where there are no stars behind it, otherwise it would be very clear it is actually see-through.

However, unlike in Vanilla, where the moon moves at the same rate as the rest of the sky, Stellar View Moon orbits around the Earth at a different pace. Due to that, the illusion for a Vanilla-like Moon would be broken, thus the Stellar View Moon instead uses two textures, one see-through texture for the halo and another solid texture for the Moon itself. 

There are two ways you can fix your issue:

1) Enable the "Use Vanilla Moon" config option (or something along those lines, can't remember how exactly I called it right now)

2) Change your resourcepack to split the Moon into a solid texture and halo texture

**Comment by SpookyTofu** [comment]
The only option that changes the moon is to disable the moon. So that most likely means there is nothing that can be done on that department. That leaves altering the textures of the resource pack. 

So I would have to take the moon texture of the resource pack, and create two separate textures. The first one with only the moon, and the second with only the halo of that moon? Would I have to rename the textures so that the mod can recognize and use the new textures?

Btw, are these the textures you are referring to? I found them inside the mod.
![moon_phases1](https://github.com/user-attachments/assets/d213fad0-de7c-4201-80f2-37d92d50df61)
![moon_halo_phases](https://github.com/user-attachments/assets/9a5a0389-5af5-4c34-a559-4550f81f36f3)

**Comment by Povstalec** [comment]
That's interesting, I thought I added that option. Well I guess I'm misremembering it or it may have been removed somewhere along the way. No matter, the next version will have a much better system for this.

Yes, those are the textures I am referring to.

**Comment by SpookyTofu** [comment]
Alright, I split the existing resource moon texture into the halo and moon textures. I renamed them to match the mod packs textures and so far the mod hasn't recognized the new textures. 
![moon_phase](https://github.com/user-attachments/assets/d134de3d-dae4-4a57-8a42-c800ba9ebdcd)
![moon_halo](https://github.com/user-attachments/assets/a50fbd79-66db-4003-9487-bcf11152e4b2)

I was able to find a third moon texture within the mod. Do you think that this one is also needed?
![moon_phases_overlay](https://github.com/user-attachments/assets/dcec263d-05f6-4dda-9422-96232da46a57)

**Comment by SpookyTofu** [comment]
Oops, didn't mean to close the comment.
Alright, I split the existing resource moon texture into the halo and moon textures. I renamed them to match the mod packs textures and so far the mod hasn't recognized the new textures.
![moon_phase](https://github.com/user-attachments/assets/b150271f-4b87-4702-8478-9967d0ae5edf)
![moon_halo](https://github.com/user-attachments/assets/dcd054d3-9acf-4e63-9df9-b8924e2aa87a)

I did find a third texture within the mod. Do you think this one is also needed?
![moon_phases_overlay](https://github.com/user-attachments/assets/00d8eef5-c570-477d-8b21-bb1432cda400)

**Comment by Povstalec** [comment]
No, it's not needed.

Where are you putting the textures?

**Comment by SpookyTofu** [comment]
After reading your comment I realized what I was doing wrong! Within the resource pack I was replacing the moon texture from the Minecraft folder. I noticed that there where folders for other mods in the resource pack to add compatibility between them. So I made a folder for Stellar View and added the textures there and it worked!
![WorkingMoon](https://github.com/user-attachments/assets/98ac7608-0538-4ce0-ad34-c59c6b4b02e2)
Thank you for the help!
